concrete/src/Mail/Service.php

```diff
@@ -73,61 +155,96 @@ public function reset()
+     * @param mixed $val the value of the parameter
      */
     public function addParameter($key, $val)
     {
         $this->data[$key] = $val;
     }
 
     /**
-     * Add a File entity as a mail attachment.
+     * Add a File entity as an attachment of the message.
      *
-     * @param File $fob File to attach
+     * @param File $file The file to attach to the message
+     * @param array $additionalHeaders Additional headers fo the MIME part. Valid values are:
+     * - encoding: the value of the Content-Transfer-Encoding header (defaults to base64)
+     * - disposition: the main value of the Content-Disposition header (defaults to attachment)
+     * - charset: the charset value of the Content-Type header
+     * - boundary: the boundary value of the Content-Type header
+     * - id: the value of the Content-ID header (without angular brackets)
+     * - description: the value of the Content-Description header
+     * - location: the value of the Content-Location header
+     * - language: the value of the Content-Language header
      */
-    public function addAttachment(File $file)
+    public function addAttachment(File $file, array $additionalHeaders = [])
     {
         $fileVersion = $file->getVersion();
         $resource = $fileVersion->getFileResource();
         $this->addRawAttachment(
             $resource->read(),
             $fileVersion->getFilename(),
-            $fileVersion->getMimeType()
+            $fileVersion->getMimeType(),
+            $additionalHeaders
         );
     }
 
     /**
      * Add a mail attachment by specifying its raw binary data.
      *
      * @param string $content The binary data of the attachemt
-     * @param string $filename The name to give to the attachment
-     * @param string $mimetype The MIME type of the attachment
+     * @param string $filename The name to give to the attachment (it will be used as the filename part of the Content-Disposition header)
+     * @param string $mimetype The MIME type of the attachment (it will be the main value of the Content-Type header)
+     * @param array $additionalHeaders Additional headers fo the MIME part. Valid values are:
+     * - encoding: the value of the Content-Transfer-Encoding header (defaults to base64)
+     * - disposition: the main value of the Content-Disposition header (defaults to attachment)
+     * - charset: the charset value of the Content-Type header
+     * - boundary: the boundary value of the Content-Type header
+     * - id: the value of the Content-ID header (without angular brackets)
+     * - description: the value of the Content-Description header
+     * - location: the value of the Content-Location header
+     * - language: the value of the Content-Language header
      */
-    public function addRawAttachment($content, $filename, $mimetype = 'application/octet-stream')
+    public function addRawAttachment($content, $filename, $mimetype = 'application/octet-stream', array $additionalHeaders = [])
```

Comment:
BC break, could we get around this by overloading that parameter and using func_get_args()  to access it?

Reply:
What about adding new methods? I think that having explicit arguments are much better for the sake of simplicity. BTW I'm really open to suggestions about the new method names :wink:

Reply:
Check out my other comment referring to symfony's policy on BC. I think they do preserve BC when it comes to adding parameters to methods. [Here's the 3v4l that shows what I mean](https://3v4l.org/GqSPa).

Reply:
What if we just added two new methods 

addAttachmentWithHeaders
addRawAttachmentWithHeaders